core-compose/src/androidTest/java/com/squareup/workflow/ui/compose/internal/ViewFactoriesTest.kt

```diff
@@ -48,14 +60,91 @@ class ViewFactoriesTest {
       WorkflowRendering(TestRendering("two"), viewEnvironment)
     }
 
-    findByText("one\ntwo").assertIsDisplayed()
+    onNodeWithText("one", useUnmergedTree = true).assertIsDisplayed()
+    onNodeWithText("two", useUnmergedTree = true).assertIsDisplayed()
+  }
+
+  @Test fun WorkflowRendering_rendersLegacyAndroidView() {
+    val viewEnvironment = ViewEnvironment(ViewRegistry(LegacyViewViewFactory))
+        .withCompositionRoot { content ->
+          Column {
+            Text("one")
+            content()
+          }
+        }
+
+    composeRule.setContent {
+      WorkflowRendering(LegacyViewRendering("two"), viewEnvironment)
+    }
+
+    onNodeWithText("one", useUnmergedTree = true).assertDoesNotExist()
+    onNodeWithTag(LegacyAndroidViewTestTag, useUnmergedTree = true).assertIsDisplayed()
+  }
+
+  @OptIn(ExperimentalLayoutNodeApi::class)
+  @Test fun WorkflowRendering_legacyAndroidViewRendersUpdates() {
+    // Our view ends up being nested pretty deeply in a bunch of ViewGroups, so iterate through
+    // them all until we find it.
+    fun AndroidOwner.findTextView() = generateSequence<View>(view as ViewGroup) {
+      when (it) {
+        is TextView -> null
+        is ViewGroup -> it.getChildAt(0)
+        else -> error("No text view!")
+      }
+    }.last() as TextView
+
+    val wrapperText = mutableStateOf("two")
+    val viewEnvironment = ViewEnvironment(ViewRegistry(LegacyViewViewFactory))
+
+    composeRule.setContent {
+      WorkflowRendering(LegacyViewRendering(wrapperText.value), viewEnvironment)
+    }
+
+    val owner = onNodeWithTag(LegacyAndroidViewTestTag, useUnmergedTree = true)
+        .fetchSemanticsNode()
+        .componentNode
+        .owner!! as AndroidOwner
+
+    // Really digging into internals here. This would work better if the accessibility
+    // inter-op when creating a LayoutNode from a View existed, so that we could continue
+    // to use the standard Compose testing API when dealing with Android views.
```

Comment:
Did you file a Compose issue for this? https://issuetracker.google.com/issues?q=componentid:612128

Reply:
I didn't. There's a TODO code comment where the update would happen at (`AndroidViewHolder.toLayoutNode()`) that would fix this. Think it's still worth filing?